Core/Src/can_handler.c

```diff
@@ -0,0 +1,158 @@
+/**
+ * @file can_handler.c
+ * @author Hamza Iqbal
+ * @brief Source file for CAN handler.  
+ * @version 0.1
+ * @date 2023-09-22
+ * 
+ * @copyright Copyright (c) 2023
+ * 
+ */
+
+#include <stdlib.h>
+#include "can.h"
+#include "can_config.h"
+
+/* Initializes the CAN handler */
+void can_handler_init()
+{
+    // Assign memory to the queues
+    can1_incoming = malloc(sizeof(msg_queue));
+    can2_incoming = malloc(sizeof(msg_queue));
+    can1_outgoing = malloc(sizeof(msg_queue));
+    can2_outgoing = malloc(sizeof(msg_queue));
+
+    initializeHashMap(&callback_map);
+
+    for(int i = 0; i < NUM_CALLBACKS; i++)
```

Comment:
So we put stuff into hash map at boot, got it. I really gotta get documentation or an explanation on how it work because me dumb. I support the hash map approach, if we index by CAN ID it should be more efficient than a ginormous LUT